src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -205,6 +208,16 @@ def visit_FencilDefinition(self, node: itir.FencilDefinition):
                 access_node = last_state.add_access(inner_name)
                 last_state.add_edge(nsdfg_node, inner_name, access_node, None, memlet)
 
+        # Create the call signature for the SDFG.
+        #  All arguments required by the SDFG, regardless if explicit and implicit, are added
+        #  as positional arguments. In the front are all arguments to the Fencil, in that
+        #  order, they are followed by the arguments created by the translation process,
+        arg_list = [str(a) for a in node.params]
+        sig_list = program_sdfg.signature_arglist(with_types=False)
+        implicit_args = set(sig_list) - set(arg_list)
+        call_params = arg_list + [ia for ia in sig_list if ia in implicit_args]
```

Comment:
I don't know how this works, so I am just wondering in which order we should present the fencil arguments (`arg_list`). Should we maybe make sure that the arguments are in the same order as they appear in `sig_list`? 
I mean:
`call_params = [fa for fa in sig_list if fa in arg_list] + [ia for ia in sig_list if ia in implicit_args]`

From the API it is not clear if `sig_list` will follow the same order as `node.params`.

Reply:
This part is extremely strange. `sig_list` is the signature of the actuall `C++` function and `arg_names` is the (ordered) list of arguments how you can call the SDFG.
When you call `CompiledSDFG._call__()` the function expects that (positional) arguments are passed in the order  specified by `arg_names` before it will then perform the call to the `C++` function it will rearanging the passed arguments.
However, changing the order of `arg_names` directly to the one given by `sig_list` has no effect because, `CompiledSDFG.__call__()` will pack everything in a `dict` before it does the processing.

So I think that the best choice is to preserving the order given by the fencil, since it that what a user would expect. However, the additional arguments, that are produced by the process could be in any order.

As a side note, the order in `sig_list` is first all array arguments then all scalar, inside these two groups they are ordered alphabetical.

Reply:
This part is extremely strange. `sig_list` is the signature of the actuall `C++` function and `arg_names` is the (ordered) list of arguments how you can call the SDFG.
When you call `CompiledSDFG._call__()` the function expects that (positional) arguments are passed in the order  specified by `arg_names` before it will then perform the call to the `C++` function it will rearanging the passed arguments.
However, changing the order of `arg_names` directly to the one given by `sig_list` has no effect because, `CompiledSDFG.__call__()` will pack everything in a `dict` before it does the processing.

So I think that the best choice is to preserving the order given by the fencil, since it that what a user would expect. However, the additional arguments, that are produced by the process could be in any order.

As a side note, the order in `sig_list` is first all array arguments then all scalar, inside these two groups they are ordered alphabetical.

Reply:
I see. Thank you for looking into this and for the explanation.